sqlite3.go

```diff
@@ -1354,9 +1360,19 @@ func (d *SQLiteDriver) Open(dsn string) (driver.Conn, error) {
 	var db *C.sqlite3
 	name := C.CString(dsn)
 	defer C.free(unsafe.Pointer(name))
+
+	// SQLite3 initialization is not thread-safe (see the prologue of sqlite3_initialize) and
+	// sqlite3_open_v2 implicitly calls sqlite3_initialize().
+	// This initCh channel contains a single dummy message so the first call performs initialization
+	// while other concurrent calls are waiting. Then the channel closed and does not work as a init barrier anymore.
+	_, ok := <-initCh
 	rv := C._sqlite3_open_v2(name, &db,
 		mutex|C.SQLITE_OPEN_READWRITE|C.SQLITE_OPEN_CREATE,
 		nil)
+	if ok {
+		close(initCh)
```

Comment:
This means you will close the channel even if `sqlite3_open_v2` were to fail, which means your init barrier will not work as expected on the next invocation.

Reply:
you are right, good catch